### PR TITLE
How to communicate between modules

### PR DESCRIPTION
**Pull request by federiva**
Adding vignette - How to Communicate between modules
Peek 2023-09-06 15-10

### CODE REVIEW

File: vignettes/how-to/communicate-between-modules.Rmd [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+we pass it down to the plot module.
+
+```R
+box::use(
+  shiny[bootstrapPage, div, moduleServer, NS, ],
+)
+
+box::use(
+  data_module = app / view / processing_data_module,
+  plot_module = app / view / plotting_module
```
Comment (jakubnowicki) [inline vignettes/how-to/communicate-between-modules.Rmd:105]: Please remove whitespaces:
```suggestion
  data_module = app/view/processing_data_module,
  plot_module = app/view/plotting_module,
```

File: vignettes/how-to/communicate-between-modules.Rmd [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+
+After we have our sibling modules ready we use them in our `main.R` function.  
+Pay special attention to the server function in which we save the output of the data processing module in a variable and
+we pass it down to the plot module.
+
+```R
+box::use(
+  shiny[bootstrapPage, div, moduleServer, NS, ],
```
Comment (jakubnowicki) [inline vignettes/how-to/communicate-between-modules.Rmd:100]:
```suggestion
  shiny[bootstrapPage, div, moduleServer, NS],
```

File: vignettes/how-to/communicate-between-modules.Rmd [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+Suppose we have a data processing module and a plotting module, both functioning as siblings, with their respective roles
+being to process and exhibit data in a given plot. These sibling modules are nested within another module, which we've
+denoted as `main.R` which is our main module within a Rhino app.
+
+![Figure 1. Diagram illustrating how a reactive value is passed between sibling modules](images/communicate_between_modules_1.png)
+
+Let's explore how the code of these modules should look like:
+
+```R
```
Comment (jakubnowicki) [inline vignettes/how-to/communicate-between-modules.Rmd:27]: For consistency, please add `box` imports.

File: vignettes/how-to/communicate-between-modules.Rmd [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    
+    # Returns the reactive
+    data_to_display
+  })
+}
+```
+
+And the plot module
+
+```R
```
Comment (jakubnowicki) [inline vignettes/how-to/communicate-between-modules.Rmd:64]: For consistency, please add `box` imports.

File: vignettes/how-to/communicate-between-modules.Rmd [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  ns <- NS(id)
+  div(
+    plotOutput(
+      inputId = ns("plot")
+    )
+  )
+)
+
+#' @params id Id of the module
+#' @params data_to_display A reactive that contains that the data that will be plotted
```
Comment (jakubnowicki) [inline vignettes/how-to/communicate-between-modules.Rmd:79]: Please split into two lines:
image

**Review by jakubnowicki — CHANGES_REQUESTED** [review]
I really like how this guide is written. I've added some minor comments in the code, but I have one larger request - can you also show a case with nested modules communication?

File: vignettes/how-to/communicate-between-modules.Rmd (hunk 1 above)
Comment (federiva) [reply vignettes/how-to/communicate-between-modules.Rmd:105]: done

File: vignettes/how-to/communicate-between-modules.Rmd (hunk 2 above)
Comment (federiva) [reply vignettes/how-to/communicate-between-modules.Rmd:100]: done

File: vignettes/how-to/communicate-between-modules.Rmd (hunk 3 above)
Comment (federiva) [reply vignettes/how-to/communicate-between-modules.Rmd:27]: done

File: vignettes/how-to/communicate-between-modules.Rmd (hunk 4 above)
Comment (federiva) [reply vignettes/how-to/communicate-between-modules.Rmd:64]: doen

File: vignettes/how-to/communicate-between-modules.Rmd (hunk 5 above)
Comment (federiva) [reply vignettes/how-to/communicate-between-modules.Rmd:79]: good catch!

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Pull request merged**